### PR TITLE
doc: streamswap spec

### PR DESCRIPTION
**Pull request by robert-zaremba**
## What is the purpose of the change

Specification for the new `x/streamswap` module.

## Brief Changelog

* spec doc as README.md file
* updated comment in streamswap proto file.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? 
      yes - new x/streamswap API. However it's not wired in this PR so I suggest to put it in a changelog once thing will be wired up.

### CODE REVIEW

File: x/streamswap/README.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+Anyone can join a sale by sending a [MsgSubscribe](https://github.com/osmosis-labs/osmosis/blob/main/proto/osmosis/streamswap/v1/tx.proto#L13) transaction.
+When doing so, the transaction author has to specify the `amount` he wants to spend in the sale.
+That `amount` will be credited from tx author and pledged to the sale.
+
+`MsgSubscribe` can be submitted at any time after the sale is created and before it's end time.
+
+From that moment, the investor will join the **token sale distribution stream**:
+
+- The distribution happens discretely in rounds. Each round is 1 second long.
+  We define: `total_rounds := (sale.end_time - sale.start_time) / round_duration`.
```
Comment (stackman27) [inline x/streamswap/README.md:80]: Why even divide if `round_duration` is always going to be 1? Also is there a reason why each round is always 1?

File: x/streamswap/README.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+- During each round we stream `round_supply := sale.token_out_supply / total_rounds`.
+- During each round each investor receives `round_supply * current_investor_pledge / total_remainding_sale_pledge` share of the `token_out`.
```
Comment (stackman27) [inline x/streamswap/README.md:82]: Is `total_remaining_sale_pledge` the same as `sale.token_out`? 

```suggestion
- During each round each investor receives `round_supply * current_investor_pledge / total_remaining_sale_pledge` share of the `token_out`.
```

File: x/streamswap/README.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  find a fair strike price.
+
+The first two mechanisms are not well suited for early stage startups, where
+the token sale price is usually defined by a founding team and can't be
+impacted by the ecosystem wisdom. False marketing actions are usually setup
+to support their initial price.
+
+The latter mechanism is not democratic - big entities can control the
+price movements or place big orders leaving smaller investors with nothing.
+
```
Comment (stackman27) [inline x/streamswap/README.md:34]: I think it'll be really helpful to add what makes streamswap more "democratic" than the rest

File: x/streamswap/README.md (hunk 1 above)
Comment (robert-zaremba) [reply x/streamswap/README.md:80]: Well, this is the definition. Currently round_duration is set to 1second, but that can change.

File: x/streamswap/README.md (hunk 3 above)
Comment (robert-zaremba) [reply x/streamswap/README.md:34]: It's explained below: anyone can join and has guarantee to received tokens.

**Pull request merged**